### PR TITLE
feat: API error messages

### LINKED ISSUE #221: Implement mechanism to show proper localized messages for errors coming from the API
After the attached API change, errors returned by the API has a structure of code and context. This is probably enough to have these errors properly translated to a human-readable message on the web client. So, this issue is about implementing this behavior on the web client that handles the API error responses and present an appropriate message to the users according to their languages.

https://github.com/techmatters/terraso-backend/pull/115

### PR DESCRIPTION
**Pull request by josebui**
### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added
- [ ] Strings are localized
- [x] Verified on mobile
- [x] Verified on desktop
- [x] Verified accessibility ([Accessibility - Web applications](https://docs.google.com/document/d/1VwPDyLqw7r_iISMgDZr1FdAmAqqOIIVIxQqsYA-HaHE/edit?usp=sharing))

### Related Issues
Fixes #221

### CODE REVIEW

**Review by paulschreiber — CHANGES_REQUESTED** [review]

File: src/localization/locales/en-US.json [hunk 1]
```diff
@@ -227,8 +228,21 @@
...
+        "error": "Failed to execute request. {{code, errorParam(label: Code)}} {{field, errorParam(label: Field)}} {{model, errorParam(label: Model)}} {{context, errorParam(label: Context)}}",
         "error_unexpected": "Unexpected error",
-        "error_request_response": "Failed to connect with Terraso API"
+        "error_request_response": "Failed to connect with Terraso API",
+        "deleteNotAllowed_base": "Not allowed to delete $t(terraso_api.models.{{model}})",
+        "deleteNotAllowed": "$t(terraso_api.deleteNotAllowed_base)",
+        "createNotAllowed_base": "Not allowed to create $t(terraso_api.models.{{model}})",
+        "createNotAllowed": "$t(terraso_api.createNotAllowed_base)",
+        "updateNotAllowed_base": "Not allowed to update $t(terraso_api.models.{{model}})",
+        "updateNotAllowed": "$t(terraso_api.updateNotAllowed_base)",
```
[the remark this answers is not in the document]
Comment (josebui) [reply src/localization/locales/en-US.json:244]: @paulschreiber For this one the code is coming from the API as `deleteNotAllowed` so we will have to programatically change it to snake cased, IMO is not worth it, waht do you think?

[the remark this answers is not in the document]
Comment (paulschreiber) [reply src/localization/locales/en-US.json:244]: If we can't change what the API returns, then we can leave as-is.

[the remark this answers is not in the document]
Comment (josebui) [reply src/localization/locales/en-US.json:244]: That makes sense, I will check the API

File: src/localization/locales/en-US.json [hunk 2]
```diff
@@ -79,7 +79,8 @@
         "view_document_title": "{{name}}",
         "members_document_title": "Members: {{name}}",
         "list_column_actions_description": "Actions",
-        "members_list_column_actions_description": "Actions"
+        "members_list_column_actions_description": "Actions",
+        "delete_not_allowed_manager_count": "$t(terraso_api.delete_not_allowed_base). There should at least be one manager."
```
[the remark this answers is not in the document]
Comment (josebui) [reply src/localization/locales/en-US.json:83]: @paulschreiber It just calls the other key, you can also pass paramters to it: you can check that here: https://www.i18next.com/translation-function/nesting

**Pull request merged**